### PR TITLE
DOC: Apply doc sprint

### PR DESCRIPTION
**Pull request by dtammaro**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
# paste output of "scripts/validate_docstrings.py <your-function-or-method>" here
# between the "```" (remove this comment, but keep the "```")

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -4828,10 +4829,11 @@ def apply(self, func, axis=0, broadcast=None, raw=False, reduce=None,
         Parameters
         ----------
         func : function
-            Function to apply to each column/row
+            Function to apply to each column/row.
         axis : {0 or 'index', 1 or 'columns'}, default 0
+            Axis along which the function is applied
```
Comment (TomAugspurger) [inline pandas/core/frame.py]: This line can have end with a period.

**Comment by datapythonista** [comment]
@dtammaro Can you please udpate the title of the pull request, and copy the output of the script in the specified section? it's difficult to review otherwise

File: pandas/core/frame.py [hunk 2]
```diff
@@ -4818,7 +4818,8 @@ def aggregate(self, func, axis=0, *args, **kwargs):
 
     def apply(self, func, axis=0, broadcast=None, raw=False, reduce=None,
               result_type=None, args=(), **kwds):
-        """Applies function along an axis of the DataFrame.
+        """
+        Apply function along an axis of the DataFrame.
```
Comment (TomAugspurger) [inline pandas/core/frame.py]: Maybe "Apply a function along an axis of the DataFrame." (add "a" before "function").

File: pandas/core/frame.py [hunk 3]
```diff
@@ -4828,10 +4829,11 @@ def apply(self, func, axis=0, broadcast=None, raw=False, reduce=None,
         Parameters
         ----------
         func : function
-            Function to apply to each column/row
+            Function to apply to each column/row.
         axis : {0 or 'index', 1 or 'columns'}, default 0
+            Axis along which the function is applied
             * 0 or 'index': apply function to each column
-            * 1 or 'columns': apply function to each row
+            * 1 or 'columns': apply function to each row.
```
Comment (TomAugspurger) [inline pandas/core/frame.py]: These bullets should both have periods, or not. I don't really have a preference, but probably best to add a period to both.

File: pandas/core/frame.py [hunk 4]
```diff
@@ -4872,12 +4874,15 @@ def apply(self, func, axis=0, broadcast=None, raw=False, reduce=None,
...
-            .. versionadded:: 0.23.0
+            .. versionadded:: 0.23.0.
 
         args : tuple
             Positional arguments to pass to function in addition to the
-            array/series
-        Additional keyword arguments will be passed as keywords to the function
+            array/series.
+
+        kwds : dictionary
```
Comment (TomAugspurger) [inline pandas/core/frame.py]: @jorisvandenbossche remind me, what's the policy here? Just `kwargs`, with no type?

Comment (TomAugspurger) [reply pandas/core/frame.py]: Though probably just `kwds` in this case.

Comment (datapythonista) [reply pandas/core/frame.py]: `**kwargs` with no type

**Comment by jreback** [comment]
@datapythonista this doc-string prob needs another look, it is pretty long and complex so might be some more changes.

**Comment by jorisvandenbossche** [comment]
This has been merged in https://github.com/pandas-dev/pandas/pull/20202

**Pull request closed without merging**